be/src/runtime/string_value.hpp

```diff
@@ -67,6 +67,32 @@ static inline int string_compare(const char* s1, int64_t n1, const char* s2, int
     return n1 - n2;
 }
 
+// This is a special compare algorithm for len(s1) == len(s2) scenarios
+// faster than generic string_compare algorithm.
+// This code assumes that the trivial cases are already handled (i.e. len(s1) != len(s2)).
+// Returns:
+//   true  if s1 == s2
+//   false if s1 != s2
+static inline bool eq_string_compare(const char* s1, const char* s2, int64_t len) {
+#ifdef __SSE4_2__
```

Comment:
the code seem simple, maybe you should try use `auto vectorized` with compiler. which can use avx2 or other simd code